livedata/src/main/java/io/getstream/chat/android/livedata/extensions/ClientExtensions.kt

```diff
@@ -28,16 +28,13 @@ fun ChatEvent.getCid(): String? {
 /**
  * cid is sometimes devent as message.cid other times as message.channel.cid
  */
-fun Message.getCid(): String? {
-
-    var cid = this.cid
-
-    if (cid.isNullOrEmpty()) {
-        cid = this.channel?.cid
+fun Message.getCid(): String =
```

Comment:
What do you think about to override the getter of `Message#cid` to do this logic and be sure it is always used on the proper way?

It could be implemented like:
```
data class Message(private val _cid: String) {
  var cid: String = _cid
    get() = field.takeUnless { it.isEmpty() } ?: channel.cid
}
```

Reply:
Mmm the problem with this, is taht we would be adding logic to a Data Class, which shouldn't have logic at all.
I usually prefer this kind of things as extended logic outside the data class (like the open-closed principle). 

Still, at this moment, we could do both. I don't mind changing, if we keep consistency.